### PR TITLE
Allow users to specify async-http-client version (1.9.3 compatible).

### PR DESCRIPTION
**Pull request by johnou**
@stephenc didn't want to merge this in without first discussing it with you, please let me know your thoughts.

### CODE REVIEW

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 1]
```diff
@@ -1113,8 +1113,19 @@ public void updateInstallation(int id, String json) {
...
-            throw new ZendeskException(e.getMessage(), e);
+        } catch (Throwable t) {
+            throw new ZendeskException(t);
+        }
+    }
+
+    private static abstract class ZendeskAsyncCompletionHandler<T> extends AsyncCompletionHandler<T> {
+        @Override
+        public void onThrowable(Throwable t) {
+            if (t instanceof IOException) {
```
Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: might be worth logging something here or throwing ZendeskException for all exceptions, super just chews it / logs debug.

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 2]
```diff
@@ -1113,8 +1113,19 @@ public void updateInstallation(int id, String json) {
                 logger.debug("Request {} {}", request.getMethod(), request.getUrl());
             }
             return client.executeRequest(request, handler);
```
Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: catching throwable here as 1.9.3 no longer throws IOException so we can support 1.7.12 (which anyway caught everything and wrapped it in IOException).

Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: @stephenc what did you need to check wrt compatibility, I would be for removing the try catch and bumping major version.

Comment (stephenc) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: checked, let's go for it

File: pom.xml [hunk 3]
```diff
@@ -71,7 +75,7 @@
     <dependency>
       <groupId>com.ning</groupId>
       <artifactId>async-http-client</artifactId>
-      <version>1.7.12</version>
+      <version>${async-http-client.version}</version>
```
Comment (stephenc) [inline pom.xml]: don't see the need for a property as we only have one dep.

File: pom.xml [hunk 4]
```diff
@@ -27,6 +27,10 @@
   <artifactId>zendesk-java-client</artifactId>
   <version>0.2.5-SNAPSHOT</version>
 
+  <properties>
```
Comment (stephenc) [inline pom.xml]: properties are only for use when you reuse the dependency version in multiple places in the same pom or where you want to test multiple versions from the cli

**Pull request merged**